scripts/pre_commit_linter.py

```diff
@@ -580,6 +580,98 @@ def _check_bad_pattern_in_file(filename, content, pattern):
     return False
 
 
+def _check_bad_pattern_in_comments(filename, content):
+    """Detects whether comments end with a period.
+
+    Args:
+        filename: str. Name of the file.
+        content: str. Contents of the file.
+
+    Returns:
+        bool. True if there is bad pattern else false.
+    """
+    regexp_comment = r'#[^\n]*[^.,\n]\n( |\t)*(#(( |\t)*[A-Z])|[^#\s]|[\n])'
+    message = 'There should be a period at the end of the comment.'
+    punctuations = ['.', ',', ';', ':', '?']
```

Comment:
Since this will be used by the other check, you'll want to pull this out as a global variable like the section in [line 100](https://github.com/bansalnitish/oppia/blob/22f0b1fcfd8085a822a8046bbfe465387435bf81/scripts/pre_commit_linter.py#L100)